internal/trace/transport.go

```diff
@@ -0,0 +1,74 @@
+// Copied and adapted from oras (https://github.com/oras-project/oras)
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+Copied and adapted from oras (https://github.com/oras-project/oras)
+*/
+
+package trace
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/notaryproject/notation-go/log"
+)
+
+// Transport is an http.RoundTripper that keeps track of the in-flight
+// request and add hooks to report HTTP tracing events.
+type Transport struct {
+	http.RoundTripper
+}
+
+func NewTransport(base http.RoundTripper) *Transport {
+	return &Transport{base}
+}
+
+// RoundTrip calls base roundtrip while keeping track of the current request.
+func (t *Transport) RoundTrip(req *http.Request) (resp *http.Response, err error) {
+	ctx := req.Context()
+	e := log.GetLogger(ctx)
+
+	e.Debugf(" Request URL: %q", req.URL)
+	e.Debugf(" Request method: %q", req.Method)
+	e.Debugf(" Request headers:")
```

Comment:
```suggestion
	e.Debugf("> Request: %q %q", req.Method, req.URL)
	e.Debugf("> Request headers:")
```

/cc @qweeah to check if it should be applied to oras.

Reply:
Updated.